plugins/formal-verification/src/org/jetbrains/kotlin/formver/conversion/ContractDescriptionConversionVisitor.kt

```diff
@@ -35,18 +38,32 @@ class ContractDescriptionConversionVisitor : KtContractDescriptionVisitor<Exp, M
         val retVar = data.returnVar.toLocalVar()
         return when (returnsEffect.value) {
             ConeContractConstantValues.WILDCARD -> BoolLit(true)
-            ConeContractConstantValues.NULL -> EqCmp(retVar, NullLit())
-            ConeContractConstantValues.NOT_NULL -> NeCmp(retVar, NullLit())
+            // TODO: extend null comparisons to other types (also non-nullable), I think it is better waiting for null to work with more types
+            ConeContractConstantValues.NULL -> EqCmp(retVar, NullableDomain.nullVal(Type.Int))
+            ConeContractConstantValues.NOT_NULL -> NeCmp(retVar, NullableDomain.nullVal(Type.Int))
```

Comment:
Since the expressions are so limited here, you can make null work for any type by using `NullableDomain.nullVal(retVar.type)`

Reply:
I think that in these cases it has to be done a distinction between ret of nullable and non-nullable type. In the case of a nullable type I have to pass its parameter type, is there an easy way to get it? because yesterday I didn't find anything

Reply:
You can check if `data.returnVar.type` is `ConvertedNullable`, but maybe a helper function in `ConvertedType` might be useful here. But I guess if you want you can leave it as Int for now and we can take another look when other nullable types work as well.